libbeat/cmd/instance/ilm.go

```diff
@@ -0,0 +1,202 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package instance
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type ilmConfig struct {
+	RolloverAlias string `config:"ilm.rollover_alias" `
+	Pattern       string `config:"ilm.pattern"`
+}
+
+var ILMPolicy = common.MapStr{
+	"policy": common.MapStr{
+		"type": "timeseries",
+		"phases": common.MapStr{
+			"hot": common.MapStr{
+				"actions": common.MapStr{
+					"rollover": common.MapStr{
+						"max_size": "25gb",
+						"max_age":  "30d",
+					},
+				},
+			},
+		},
+	},
+}
+
+const (
+	ILMPolicyName     = "beats-default-policy"
+	ILMDefaultPattern = "0001"
+)
+
+// Build and return a callback to load ILM write alias
+func (b *Beat) writeAliasLoadingCallback() (func(esClient *elasticsearch.Client) error, error) {
+	callback := func(esClient *elasticsearch.Client) error {
+		if b.Config.ILM == nil {
+			b.Config.ILM = common.NewConfig()
+		}
+
+		config, err := getILMConfig(b)
+		if err != nil {
+			return err
+		}
+
+		firstIndex := fmt.Sprintf("%s-%s", config.RolloverAlias, config.Pattern)
+		// Create write alias
+		status, b, err := esClient.Request("HEAD", "/_alias/"+config.RolloverAlias, "", nil, nil)
+		if err != nil && status != 404 {
+			logp.Err("Failed create write alias: %s: %+v", err, string(b))
+			return errors.Wrap(err, "failed to check for write alias")
+		}
+		if status == 200 {
+			logp.Info("Write alias already exists")
+			return nil
+		}
+
+		body := common.MapStr{
+			"aliases": common.MapStr{
+				config.RolloverAlias: common.MapStr{
+					"is_write_index": true,
+				},
+			},
+		}
+
+		// Create write alias
+		_, res, err := esClient.Request("PUT", "/"+firstIndex, "", nil, body)
```

Comment:
As mentioned before, this is open for raise conditions, if multiple beats are started and checking for the alias the same time.

Reply:
fixed now by checking for 400.